lib/practice_exercises.rb

```diff
@@ -1,13 +1,34 @@
+# Kareha Agesa
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(1/2n) = O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^A-Za-z0-9]/, '')   # Space: O(n)?, Time O(n)?
+  length = string.length                              # Space: 0(1), Time O(n)
+  (string.length/2).times do |i|                      # Time: O(1/2n)
+    return false if string[i] != string[length-1]
+    length -= 1                                       # Space: 0(1)
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)?
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
👍 This works, but the time complexity is O(n * m)  where m is the length of the shortest string.